napps/kytos/of_l2ls/main.py

```diff
@@ -51,6 +52,9 @@ def handle_packet_in(self, event):
 
         switch.update_mac_table(ethernet.source, in_port)
 
+        if packet_in.reason != PacketInReason.OFPR_NO_MATCH:
+            return
```

Comment:
One may have a NApp that filters some packets (for any kind of special treatment) and desire l2ls behavior for some other packets. Some of those filters may need to send the 'forwardable' 'packet_in's to the controller before checking all rules on a flow table (and coming to a table_miss). 
Ignoring those may reduce the functionality of l2ls without any real benefit.